src/error/metadata/mod.rs

```diff
@@ -59,26 +59,26 @@ impl From<&mut anyhow::Error> for Metadata {
                         valid_variants: valid_variants.clone(),
                         frontend_url_root: frontend_url_root.clone(),
                     }),
-                    None,
+                    Some(Code::E008),
                 ),
                 RoverClientError::NoSubgraphInGraph {
                     invalid_subgraph: _,
                     valid_subgraphs,
                 } => (
                     Some(Suggestion::ProvideValidSubgraph(valid_subgraphs.clone())),
-                    None,
+                    Some(Code::E009),
                 ),
                 RoverClientError::NoService { graph: _ } => {
-                    (Some(Suggestion::CheckGraphNameAndAuth), None)
-                }
-                RoverClientError::AdhocError { msg: _ }
-                | RoverClientError::GraphQl { msg: _ }
-                | RoverClientError::IntrospectionError { msg: _ }
-                | RoverClientError::ClientError { msg: _ } => (None, None),
-                RoverClientError::InvalidKey => (Some(Suggestion::CheckKey), None),
-                RoverClientError::MalformedKey => (Some(Suggestion::ProperKey), None),
+                    (Some(Suggestion::CheckGraphNameAndAuth), Some(Code::E010))
+                },
+                RoverClientError::AdhocError { msg: _ } => (None, Some(Code::E011)),
```

Comment:
We should maybe remove `AdhocError` from the API we expose...We use it in a few places but they could all be replaced with real types. We def shouldn't have an error code for it since it's just a completely random string with no extra meaning